.github/workflows/3-last-step.yml

```diff
@@ -0,0 +1,137 @@
+name: Step # replace number with last step number
+
+on:
+  workflow_dispatch:
+  # Common event triggers: pull_request, push, issues, issue_comment (feel free to experiment)
+  # Docs: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows
+  # Example:
+  # pull_request:
+  #   branches:
+  #     - main
+  #   types:
+  #     - closed
+
+permissions:
+  contents: write
+  actions: write
+  issues: write
+
+env:
+  REVIEW_FILE: ".github/steps/x-review.md"
+
+jobs:
+  find_exercise:
+    name: Find Exercise Issue
+    uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@v0.6.0
+
+  check_step_work:
+    name: Check step work
+    runs-on: ubuntu-latest
+    needs: [find_exercise]
+    env:
+      ISSUE_URL: ${{ needs.find_exercise.outputs.issue-url }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Get response templates
+        uses: actions/checkout@v4
+        with:
+          repository: skills/exercise-toolkit
+          path: exercise-toolkit
+          ref: v0.6.0
+
+      - name: Update comment - checking work
+        run: |
+          gh issue comment "$ISSUE_URL" \
+            --body-file exercise-toolkit/markdown-templates/step-feedback/checking-work.md \
+            --edit-last
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      # START: Check practical exercise
+
+      # replace-me: Example of a grading check. You can add more checks as needed.
+      - name: Check for keyphrase in README.md
+        id: check-for-keyphrase
+        continue-on-error: true
+        uses: skills/action-keyphrase-checker@v1
+        with:
+          text-file: README.md
+          keyphrase: Installation guide
+
+      - name: Build message - step results
+        id: build-message-step-results
+        uses: skills/action-text-variables@v2
+        with:
+          template-file: exercise-toolkit/markdown-templates/step-feedback/step-results-table.md
+          template-vars: |
+            step_number: 1
+            passed: ${{ !contains(steps.*.outcome, 'failure') }}
+            results_table:
+              - description: "Checked if README.md file exists"
+                passed: ${{ steps.check-file-exists.outcome == 'success' }}
+              - description: "Checked for Installation guide in README.md"
+                passed: ${{ steps.check-for-keyphrase.outcome == 'success' }}
+          # Add more checks as needed
+
+      # END: Check practical exercise
+
+      - name: Fail job if not all checks passed
+        if: contains(steps.*.outcome, 'failure')
+        run: exit 1
+
+      - name: Create comment - step finished - final review next
```

Comment:
Fyi @arilivigni this should be moved to "post_review_content" so if the last step does not have a `check_step_work` job the lesson-review is still included.

Same scenario you noticed when congratulating step finish was missing

Reply:
I will do it in my next PR as I will rewrite this to use grantbirki/comment action